statsmodels/gam/gam.py

```diff
@@ -0,0 +1,367 @@
+from __future__ import division
+
+__author__ = 'Luca Puggini: <lucapuggio@gmail.com>'
+__date__ = '08/07/15'
+
+import numpy as np
+import scipy as sp
+from statsmodels.discrete.discrete_model import Logit
+from scipy.stats import chi2
+from statsmodels.genmod.generalized_linear_model import GLM, GLMResults, GLMResultsWrapper, lm, _check_convergence
+from statsmodels.tools.sm_exceptions import PerfectSeparationError
+from statsmodels.gam.gam_penalties import MultivariateGamPenalty
+
+
+## this class will be later removed and taken from another push
+class PenalizedMixin(object):
+    """Mixin class for Maximum Penalized Likelihood
+    TODO: missing **kwds or explicit keywords
+    TODO: do we really need `pen_weight` keyword in likelihood methods?
+    """
+
+    def __init__(self, *args, **kwds):
+        super(PenalizedMixin, self).__init__(*args, **kwds)
+        penal = kwds.pop('penal', None)
+        # I keep the following instead of adding default in pop for future changes
+        if penal is None:
+            print("Define a penalty")
```

Comment:
It looks like `loglike` will break if this is allowed.  Maybe this should be a `ValueError`?